alchemy/test/aws/control/proxy.test.ts

```diff
@@ -0,0 +1,166 @@
+      bucket = await AWS.S3.Bucket(testId, {
+        BucketName: testId,
+        VersioningConfiguration: {
+          Status: "Enabled",
+        },
+      });
+
+      expect(bucket.BucketName).toEqual(testId);
+      expect(bucket.VersioningConfiguration.Status).toEqual("Enabled");
+
+      // Update the bucket configuration
+      bucket = await AWS.S3.Bucket(testId, {
+        BucketName: testId,
+        VersioningConfiguration: {
+          Status: "Suspended",
+        },
+      });
+
+      expect(bucket.BucketName).toEqual(testId);
+      expect(bucket.VersioningConfiguration.Status).toEqual("Suspended");
+    } finally {
+      // Clean up
+      await destroy(scope);
+    }
+  });
+
+  test("proxy returns same handler for same resource type", async (scope) => {
+    const handler1 = AWS.S3.Bucket;
+    const handler2 = AWS.S3.Bucket;
+    expect(handler1).toBe(handler2);
+  });
+
+  test("proxy constructs correct type names", async (scope) => {
+    const testRoleId = `${BRANCH_PREFIX}-test-role`;
+    try {
+      const role = await AWS.IAM.Role(testRoleId, {
+        RoleName: testRoleId,
+        AssumeRolePolicyDocument: {
+          Version: "2012-10-17",
+          Statement: [
+            {
+              Effect: "Allow",
+              Principal: {
+                Service: "lambda.amazonaws.com",
+              },
+              Action: "sts:AssumeRole",
+            },
+          ],
+        },
+      });
+
+      expect(role.RoleName).toEqual(testRoleId);
+    } finally {
+      await destroy(scope);
+    }
+  });
+
+  test("handles invalid bucket configuration", async (scope) => {
+    try {
+      // Attempt to create a bucket with invalid configuration
+      await AWS.S3.Bucket(testId, {
+        BucketName: "invalid/bucket/name", // Invalid character in bucket name
+        VersioningConfiguration: {
+          Status: "InvalidStatus", // Invalid versioning status
+        },
+      });
+      throw new Error("Should have thrown an error");
+    } catch (error: any) {
+      expect(error.message).toContain("Bad Request");
+    } finally {
+      await destroy(scope);
+    }
+  });
+
+  test("create and delete DynamoDB table", async (scope) => {
+    const tableId = `${BRANCH_PREFIX}-test-table`;
+    try {
+      const table = await AWS.DynamoDB.Table(tableId, {
+        TableName: tableId,
+        AttributeDefinitions: [
+          {
+            AttributeName: "id",
+            AttributeType: "S",
+          } as const,
+        ],
+        KeySchema: [
+          {
+            AttributeName: "id",
+            KeyType: "HASH",
+          } as const,
+        ],
+        ProvisionedThroughput: {
+          ReadCapacityUnits: 1,
+          WriteCapacityUnits: 1,
+        },
+      });
+
+      expect(table.TableName).toEqual(tableId);
+      expect(table.KeySchema).toHaveLength(1);
+      // @ts-expect-error TODO: fix this
+      expect(table.KeySchema[0].AttributeName).toEqual("id");
```

Comment:
@sam-goodwin should we make `{}` `Record<string, unknown>`?

<img width="1652" alt="image" src="https://github.com/user-attachments/assets/5e7c397a-b35b-42aa-b91c-db909122c4f5" />

Reply:
I noticed that AssumeRolePolicyDocument is also `{}`. `Record<string, unknown>` would be better but I also wonder if that type is not properly documented